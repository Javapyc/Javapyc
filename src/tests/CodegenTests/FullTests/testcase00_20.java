class Main {
	public static void main(String[] args) {
		LinkedList root = new LinkedList();
		int waste = root.initialize();
		
		FibLoop fil = new FibLoop();
		waste = fil.creator(20, root);
		waste = fil.printor(20, root);
		
		FactLoop fal = new FactLoop();
		waste = fal.creator(40, root);
		waste = fal.printor(40, root);
		
		if(true) {
			int x = 0;
		} else {
			int y = 0;
		}
	}
}

class INeedAForLoop {
	boolean flag;
	
	public int f(int n) {
		return 0;
	}
	
	public int creator(int n, LinkedList root) {
		if(n >= 0) {
			int waste = root.set(n, this.f(n));
			waste = this.creator(n - 1, root);
		} else {
			int waste = 0;
		}
		return 0;
	}
	
	public int printor(int n, LinkedList root) {
		if(n >= 0) {
			System.out.println(root.get(n));
			int waste = this.printor(n - 1, root);
		} else {
			int waste = 0;
		}
		return 0;
	}
}

class FibLoop extends INeedAForLoop {
	public int fibhelper(int n, int next, int last) {
		int result = 0;
		if(n == 0)
			result = last;
		else
			result = this.fibhelper(n - 1, next + last, next);
		return result;
	}

	public int f(int n) {
		flag = true;
		return this.fibhelper(n, 1, 0); 
	}
}

class FactLoop extends INeedAForLoop {
	public int facthelper(int n, int last) {
		int result = 0;
		if(n == 0)
			result = last;
		else
			result = this.facthelper(n - 1, n * last);
		return result;
	}

	public int f(int n) {
		flag = true;		
		return this.facthelper(n, 1); 
	}
}

class LinkedList {
	int here;
	LinkedList next;
	boolean nextExists;

	// My own constructor contract
	public int initialize() {
		here = 0;
		next = null;
		return 1;
	}
	
	public int set(int index, int value) {
		int result = 0;
		if(index == 0) {
			here = value;
			result = value;
		} else {
			// Dynamic expansion
			if(next != null) {
				result = next.set(index - 1, value);
			} else {
				next = new LinkedList();
				int waste = next.initialize();
				result = next.set(index - 1, value);
			}
		}
		return result;
	}
	
	public int get(int index) {
		int result = 0;
		if(index == 0) {
			result = here;
		} else {
			if(next != null) 
				result = next.get(index - 1);
			else {
				int waste = 0;
			}
		}
		return result;
	}	
}
