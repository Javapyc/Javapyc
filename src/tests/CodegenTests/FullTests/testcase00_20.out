6765
4181
2584
1597
987
610
377
233
144
89
55
34
21
13
8
5
3
2
1
1
0
0
0
0
0
0
0
0
-2147483648
-2147483648
738197504
1409286144
-1241513984
-1375731712
1484783616
-1853882368
2076180480
-775946240
862453760
-522715136
-1195114496
-2102132736
109641728
-898433024
-288522240
2004189184
2004310016
1278945280
1932053504
479001600
39916800
3628800
362880
40320
5040
720
120
24
6
2
1
1
